src/map/clif.cpp

```diff
@@ -13937,31 +13937,41 @@ void clif_parse_RemovePartyMember( int32 fd, map_session_data* sd ){
 /// 07d7 <exp share rule>.L <item pickup rule>.B <item share rule>.B (CZ_GROUPINFO_CHANGE_V2)
 void clif_parse_PartyChangeOption(int32 fd, map_session_data *sd)
 {
-	struct party_data *p;
-	int32 i,expflag;
-	int32 cmd = RFIFOW(fd,0);
-	struct s_packet_db* info = &packet_db[cmd];
+	if( sd == nullptr ){
+		return;
+	}
 
-	if( !sd->status.party_id )
+	if( sd->status.party_id == 0 ){
 		return;
+	}
+
+	party_data* party = party_search( sd->status.party_id );
 
-	p = party_search(sd->status.party_id);
-	if( p == nullptr )
+	if( party == nullptr ){
 		return;
-	ARR_FIND( 0, MAX_PARTY, i, p->data[i].sd == sd );
-	if( i == MAX_PARTY )
-		return; //Shouldn't happen
-	if( !p->party.member[i].leader )
+	}
+
+	if( !party_isleader( sd ) ){
```

Comment:
Some parts could possibly be removed
- `if( sd->status.party_id == 0 ){` is also in `party_search`
- `party_search` is also used in `party_isleader`

Also should we accept negative value for party_id? (`int32 party_id` in `struct mmo_charstatus`) shouldn't it be `if( sd->status.party_id <= 0 ){` ?